### PR TITLE
created LoggableInterface and added it's implimentation in Article

### PR DESCRIPTION
**Pull request by batare1ka**

### CODE REVIEW

File: app/Http/Middleware/LogActivityMiddleware.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+use App\Services\DebugRequestActivityLogger;
+use App\Services\DummyRequestActivityLogger;
+use App\Services\ProductionRequestActivityLogger;
+use Closure;
+use Illuminate\Http\Request;
+
+class LogActivityMiddleware
+{
+    private DummyRequestActivityLogger $logger;
+    private DebugRequestActivityLogger $debug_logger;
```
Comment (egorbwork) [inline app/Http/Middleware/LogActivityMiddleware.php]: Won't be accepted. It was not requested.

File: app/Services/ProductionRequestActivityLogger.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace App\Services;
+use Illuminate\Http\Request;
+
+class ProductionRequestActivityLogger extends AbstractRequestActivityLogger
+{
+    protected function collectRequestData(Request $request): array
+    {
+        return [
+            "name" => $request->input('name'),
```
Comment (egorbwork) [inline app/Services/ProductionRequestActivityLogger.php]: You should not log input fields in production logger, just important information about request.

**Review by egorbwork — CHANGES_REQUESTED** [review]
Can accept in this state, multiple changes are required.

**Review by egorbwork — APPROVED** [review]
Now is acceptable.

File: app/Services/ProductionRequestActivityLogger.php (hunk 2 above)
Comment (batare1ka) [reply app/Services/ProductionRequestActivityLogger.php]: corrected

File: app/Http/Middleware/LogActivityMiddleware.php (hunk 1 above)
Comment (batare1ka) [reply app/Http/Middleware/LogActivityMiddleware.php]: changed

**Pull request merged**